.github/workflows/ci.yml

```diff
@@ -70,13 +66,11 @@ jobs:
         with:
           node-version: ${{ matrix.node-version }}
 
-      - name: Install
-        run: npm install
+      - name: Install with legacy peer deps
+        run: npm install --legacy-peer-deps
+      
+      - name: Compile TS
+        run: tsc lib/**/*.ts
```

Comment:
Udit may give this feedback later, but I probably would have just kept it vanilla JS. Compilation step doesn't benefit us a whole heck of a lot and could introduce some issues!